dissect/target/plugins/os/unix/linux/proc.py

```diff
@@ -0,0 +1,718 @@
+                else:
+                    self.entry = target.fs.get(fsutil.join(proc_root, str(pid)))
+            except FileNotFoundError:
+                raise ProcessLookupError(f"Process with PID {pid} could not be found on target: {target}")
+
+            self._stat_file = self._parse_proc_stat_entry()
+            self.name = self._process_name
+
+    @internal
+    def _parse_proc_status_entry(self) -> dict:
+        """Internal function to parse the contents of /proc/[pid]/status."""
+        status = self.get("status").open()
+        status_dict = {}
+
+        for line in status.readlines():
+            key, value = line.split(b":", maxsplit=1)
+            key = key.decode().lower().strip()
+            value = value.decode().strip()
+            status_dict[key] = value
+
+        return status_dict
+
+    def _parse_proc_stat_entry(self) -> dict:
+        """Internal function to parse the contents of /proc/[pid]/stat."""
+        status = self.get("stat").open().readline()
+        status_dict = {}
+
+        # The process name exists between parentheses in the second field.
+        # Since the name can be arbitrary we have to find the first and last parentheses.
+        start_name, end_name = status.find(b"("), status.rfind(b")")
+
+        head = status[:start_name]
+        tail = status[end_name:]
+        name = status[start_name + 1 : end_name]  # noqa: E203
+        status = head + tail
+
+        for idx, part in enumerate(status.split()):
+            try:
+                part = int(part.decode())
+            except ValueError:
+                part = part.decode()
+
+            status_dict[proc_stat_name[idx]] = part
+
+        status_dict["comm"] = name
+
+        return status_dict
+
+    def _parse_environ(self) -> Iterator[Environ]:
+        """Internal function to parse entries in /proc/[pid]/environ"""
+        # entries in /proc/<pid>/environ are null-terminated
+        environ = self.get("environ").open()
+        lines = environ.read().split(b"\x00")
+
+        for line in lines:
+            if line == b"":
+                # Skip empty line
+                continue
+            try:
+                variable, contents = line.split(b"=", maxsplit=1)
+            except ValueError:
+                # Covention tells us that variable names and values are split on '='
+                # in practice this is not always the case.
+                variable = None
+                contents = line
+
+            yield Environ(variable=variable, contents=contents)
+
+    @property
+    def _boottime(self) -> int:
+        """Returns the boot time of the system.
+
+        Used internally to determine process start- and runtimes.
+        """
+        for line in self.target.fs.get("/proc/stat").open().readlines():
+            if not line.startswith(b"btime"):
+                continue
+
+            return int(line.split()[1])
+
+    @internal
```

Comment:
```suggestion
    @internal
```
As this is not a `Plugin` subclass, this annotation is not needed.